source/FlareComponent.js

```diff
@@ -1,372 +1,411 @@
-			let desiredAnimationIndex = animationName !== undefined ? animations.findIndex( animation => animation._Name === animationName ) : -1;
-			if(desiredAnimationIndex < actorInstance.animations.length && desiredAnimationIndex !== this._RuntimeAnimationIndex)
-			{
-				if(actorInstance)
-				{
-					actorInstance.dispose(graphics);	
-				}
-				actorInstance = this._ActorInstance = actor.makeInstance();
-				actorInstance.initialize(graphics);
-
-				this._RuntimeAnimationIndex = desiredAnimationIndex;
-				this._RuntimeAnimation = desiredAnimationIndex !== -1 ? actorInstance.animations[desiredAnimationIndex] : null;
-				this._AnimationTime = 0;
-			}
-
-			let runtimeAnimation = this._RuntimeAnimation;
-			if(runtimeAnimation)
-			{
-				let animationTime;
-				if(this._OverrideAnimationPosition !== null)
-				{
-					animationTime = this._AnimationTime = this._OverrideAnimationPosition * runtimeAnimation._Duration;
-				}
-				else
-				{
-					if(advanceSpeed > 0)
-					{
-						this._AnimationTime += advanceSpeed * elapsed * (this.props.playbackSpeed || 1.0);
-						if(runtimeAnimation.loop)
-						{
-							this._AnimationTime %= runtimeAnimation._Duration;
-						}
-						else if(this._AnimationTime > runtimeAnimation._Duration)
-						{
-							this._AnimationTime = runtimeAnimation._Duration;
-							this.props.onPaused && this.props.onPaused();
-						}
-					}
-					animationTime = this._AnimationTime;
-				}
-				let animationPosition = animationTime / runtimeAnimation._Duration;
-				if(animationPosition !== this._LastReportedAnimationPosition)
-				{
-					this._LastReportedAnimationPosition = animationPosition;
-					this.props.onPositionChanged && this.props.onPositionChanged(animationPosition);
-				}
-				runtimeAnimation.apply(animationTime, actorInstance, 1.0);
-			}
-			actorInstance.advance(elapsed);
-			actorInstance.draw(graphics);
-		}
-		graphics.flush();
-
-		if(advanceSpeed > 0)
-		{
-			this._IsRendering = true;
-			window.requestAnimationFrame(() => this.advance());
-		}
-		else
-		{
-			this._IsRendering = false;
-		}
-	}
-    
-    componentDidUpdate()
-    {
-        this.updateSize();
+    if (advanceSpeed > 0) {
+      this._IsRendering = true;
+      window.requestAnimationFrame(() => this.advance());
+    } else {
+      this._IsRendering = false;
     }
+  }
 
-    render()
-    {
-        const {setCanvasRef, props, state} = this;
-        const {width, height} = props;
-        const {hasActor} = state;
-        return <div style={{position:"absolute", width: width + "px", height: height + "px"}}>
-            {hasActor ? null : <Logo width="100%" height="100%" style={{position:"absolute"}}/>}
-            <canvas ref={setCanvasRef} style={{position:"absolute",width: width + "px", height: height + "px", display: hasActor ? null : "none"}}></canvas>
-            </div>;
+  _getActorArtboard(actor, artboardName) {
+    if (artboardName) {
+      return actor.getArtboard(artboardName).makeInstance();
```

Comment:
Just be aware this can throw if the artboard isn't found. For 'get' methods like this we prefer to return null instead of throwing (like Artboard.getAnimation does), but we're open to suggestions :)